xbmc/filesystem/PluginDirectory.cpp

```diff
@@ -509,10 +511,17 @@ bool CPluginDirectory::WaitOnScriptResult(const std::string &scriptPath, int scr
     if (!m_fetchComplete.WaitMSec(20))
     {
       CScriptObserver scriptObs(scriptId, m_fetchComplete);
-      if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
+      CGUIWindowManager& wm = CServiceBroker::GetGUI()->GetWindowManager();
+      CGUIDialogProgress* progress = nullptr;
+      if (wm.IsModalDialogTopmost(WINDOW_DIALOG_PROGRESS))
+        progress = wm.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS);
+      if (progress != nullptr)
       {
-        m_cancelled = true;
+        if (!progress->WaitOnEvent(m_fetchComplete, 20))
+          m_cancelled = true;
       }
+      else if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
+        m_cancelled = true;
       scriptObs.Abort();
```

Comment:
Can you sprinkle some whitespace above? Long stanzas are harder to read. Use your discretion, but here's how I would break it up to better understand the code:

```c++
CScriptObserver scriptObs(scriptId, m_fetchComplete);

CGUIWindowManager& wm = CServiceBroker::GetGUI()->GetWindowManager();

CGUIDialogProgress* progress = nullptr;
if (wm.IsModalDialogTopmost(WINDOW_DIALOG_PROGRESS))
  progress = wm.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS);

if (progress != nullptr)
{
  if (!progress->WaitOnEvent(m_fetchComplete, 20))
    m_cancelled = true;
}
else if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
  m_cancelled = true;

scriptObs.Abort();
```

Now, as I read the code, I can reason about individual stanzas much better:

```c++
CGUIWindowManager& wm = CServiceBroker::GetGUI()->GetWindowManager();
```
*OK, wm is an alias for the following block*

```c++
CGUIDialogProgress* progress = nullptr;
if (wm.IsModalDialogTopmost(WINDOW_DIALOG_PROGRESS))
  progress = wm.GetWindow<CGUIDialogProgress>(WINDOW_DIALOG_PROGRESS);
```

*This stanza is about creating and assigning the variable `progress`*

```c++
if (progress != nullptr)
{
  if (!progress->WaitOnEvent(m_fetchComplete, 20))
    m_cancelled = true;
}
else if (!CGUIDialogBusy::WaitOnEvent(m_fetchComplete, 200))
  m_cancelled = true;
```

*Here's a mass of logic and synchronization, I need some time to digest just this*

```c++
scriptObs.Abort();
```

*First line created `scriptObs`, so last line using it isn't unexpected*